src/browser_action/vitals.js

```diff
@@ -153,9 +153,9 @@
     }) => {
       if (enableOverlay === true && overlayClosedForSession == false) {
         // Overlay
-        const overlayElement = document.getElementById('web-vitals-extension-overlay');
+        let overlayElement = document.getElementById('web-vitals-extension-overlay');
         if (overlayElement === null) {
-          const overlayElement = document.createElement('div');
+          overlayElement = document.createElement('div');
```

Comment:
The first time this runs `overlayElement` is null.

By having the outer `overlayElement` as a `const` we can't redefine it, so this code creates a duplicate variable locally to the `if`. The outer `overlayElement` is still null.

This causes problems later when we try to close this null element in the event handler in line 174 and it throws the error as can;t call `remove()` on null. Let's just set as a `let` and reuse it so the event handler can reference the newly created overlay.

Reply:
Is there really a case when it was not removed from a page? It seems like you actually always creating it and then destroying with `.remove()` method. So selecting it in a first place seems strange.

Reply:
The `drawOverlay` is called each time the metrics change. So each time is has to check it's already drawn (i.e. does the selector exist?) and then if it doesn't exist, then create it and update the contents with the updated metrics, or if it already exists, just update the contents with the updated metrics.

There's probably more efficient ways to code some of this (store the overlayElement in a top level variable so you don't need to select it with each update?), but need to consider if that would cause any other issues. The PR just fixes the bug with the current implementation.

Reply:
I wonder then is this create an performance impact? Because it update the whole content (`.innerHTML = ...`) like as if you start with good metrics, then when HUD is appeared during redrawing it will degrade page performance and lower the metrics.

Reply:
I think the impact is small. The HUD is not the most complicated piece of HTML and the alternative is several query selectors that select each piece to update (the pass/fail icon, the metric). Plus the metrics don't update that often.